logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt

```diff
@@ -3,15 +3,45 @@ package com.wire.kalium.logic.data.conversation
 import com.wire.kalium.logic.data.id.PlainId
 import com.wire.kalium.logic.data.id.QualifiedID
 import com.wire.kalium.logic.data.id.TeamId
+import com.wire.kalium.logic.data.publicuser.model.OtherUser
 import com.wire.kalium.logic.data.user.User
 import com.wire.kalium.logic.data.user.UserId
-
 typealias ConversationId = QualifiedID
 
 data class Conversation(val id: ConversationId, val name: String?, val type: Type, val teamId: TeamId?) {
     enum class Type { SELF, ONE_ON_ONE, GROUP }
 }
 
+sealed class ConversationDetails(val conversation: Conversation) {
+
+    class Self(conversation: Conversation) : ConversationDetails(conversation)
+
+    class OneOne(
+        conversation: Conversation,
+        val otherUser: OtherUser,
+        val connectionState: ConnectionState,
+        val federationStatus: FederationStatus,
+        val legalHoldStatus: LegalHoldStatus
+    ) : ConversationDetails(conversation) {
+        enum class ConnectionState {
+            // The other user has sent a connection request to this one
+            INCOMING,
+
+            // This user has sent a connection request to another user
+            OUTGOING,
+
+            // The connection is complete and the conversation is in its normal state
+            ACCEPTED
+        }
+    }
+
+    class Group(conversation: Conversation) : ConversationDetails(conversation)
+
+    enum class FederationStatus {
+        NONE, GUEST, EXTERNAL
```

Comment:
What did you base this enum on? Being a guest or external is not dependent on federation.

Reply:
I think this is relevant to `User` not `Conversation`

Reply:
I think I'll take this out for now then. This can be taken in another PR in the future.


The idea was to provide an easy way to display an indication if guests or external members are present in the conversation.